tests/unit/spam/test_safelist.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2022 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Test safelist."""
+
+import os
+import pytest
+import tempfile
+
+from zenodo.modules.spam.safelist import DomainSafelist
+
+
+def test_forbidden_list_search(tmp_spam_domains_safelist_file):
```

Comment:
Given the other changes proposed for the overall class structure, I would just keep one test for the utility class itself. 

I don't see any "E2E" style tests anywhere that directly use `current_domain_forbidden_list` and `current_domain_safelist`, in which case I'm not sure if the `conftest.py` additions are necessary also.